### PR TITLE
WooCommerce: Enable refunds on orders

### LINKED ISSUE #15672: WooCommerce Orders: Refunding payment
Store owner should be able to refund a customer

### PR DESCRIPTION
**Pull request by ryelle**
This PR splits the OrderDetails component into `OrderDetailsTable` and `OrderRefundCard`. In `OrderRefundCard`, I've added a refund modal so that store owners can refund customers.

Screenshot of the modal, where a user can refund based on items quantities, shipping cost, or enter a custom refund amount:

screen shot 2017-06-27 at 7 43 04 pm

If they try to refund more than the cost of the order, they see an error:

screen shot 2017-06-27 at 8 07 17 pm

If the refund is successful, there's a global notice after the modal closes (if the refund fails on the server side, it's an error notice), and the order details update with the new refund amount.

screen shot 2017-06-27 at 8 18 54 pm

Left to do:

- ~Entering in a custom refund amount is broken right now, either I'll fix this or convert it to read only.~ Custom amount has been switched to read-only, formatting the price breaks the input
- ~The credit card display needs to be hooked up.~ See https://github.com/Automattic/wp-calypso/pull/15589#issuecomment-311780309, this will be coming out and replaced with payment method info
- ~A complete refund should toggle the order status to refunded.~ Does this automatically 🎉

**To test**

1. Make sure you have [the latest master API plugin](https://github.com/woocommerce/wc-api-dev) on your site
2. Visit your orders list `/store/orders/:site`
3. Click into an order
4. Click the "Submit Refund" button
5. Create a refund, try with various refund amounts (0, higher than total, partial refund, exact...)

Run `npm test` to test the new action/reducer/selector (for tracking refund requests).

Fixes #15672

### CODE REVIEW

File: client/extensions/woocommerce/app/order/order-details-table.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	renderRefundValue = () => {
+		const { order, translate } = this.props;
+		const refundValue = order.refunds.length ? this.getRefundedTotal( order ) : false;
+		if ( ! refundValue ) {
+			return null;
+		}
+
+		return (
+			<div className="order__details-total-refund">
+				<div className="order__details-totals-label">{ translate( 'Refunded' ) }</div>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/order/order-details-table.js]: Hi! I've found a possible matching string that has already been translated **37** times:
`translate( 'Refunded To' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/woo-orders-refund)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by ryelle** [comment]
So, it turns out we don't know just from the API whether a payment method supports automated refunds. For example, check payments don't support automated refunds, but PayPal does. Right now if a store owner tries to refund an order paid by check, it will error. It's fine if the customer paid with paypal, though.

The credit card display here also doesn't work, the automated refund should happen from the payment gateway - so orders paid via PayPal would refund from paypal, stripe from stripe, etc. This could be "Refunding payment from PayPal" or "Refunding payment from Stripe" instead.

File: client/extensions/woocommerce/app/order/order-refund-card.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+			return;
+		} else if ( this.state.refundTotal <= 0 ) {
+			this.setState( { errorMessage: translate( 'Refund must be greater than zero.' ) } );
+			return;
+		}
+		this.toggleDialog();
+		const refundObj = {
+			amount: this.state.refundTotal + '', // API expects a string
+			reason: this.state.refundNote,
+			// api_refund
```
Comment (ryelle) [inline client/extensions/woocommerce/app/order/order-refund-card.js]: This line will turn on/off automated refunds, depending on whether this payment method supports it

**Comment by ryelle** [comment]
I've just realized I want to change my approach here for the state, so I'm putting this back into in-progress.

File: client/extensions/woocommerce/app/order/order-refund-card.js (hunk 2 above)
Comment (ryelle) [inline client/extensions/woocommerce/app/order/order-refund-card.js]: This has been handled in #15702 (which builds on this PR)

**Pull request merged**